templates/boarding-pass/pdf_boarding_pass.go

```diff
@@ -0,0 +1,171 @@
+/*
+ * This example showcases the usage of creator templates by creating a sample
+ * boarding pass document.
+ *
+ * Run as: go run pdf_boarding_pass.go
+ */
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"log"
+	"os"
+	"text/template"
+
+	"github.com/boombuler/barcode"
+	"github.com/boombuler/barcode/qr"
+	"github.com/unidoc/unipdf/v3/common"
+	"github.com/unidoc/unipdf/v3/common/license"
+	"github.com/unidoc/unipdf/v3/core"
+	"github.com/unidoc/unipdf/v3/creator"
+	"github.com/unidoc/unipdf/v3/model"
+)
+
+func init() {
+	// Make sure to load your metered License API key prior to using the library.
+	// If you need a key, you can sign up and create a free one at https://cloud.unidoc.io.
+	err := license.SetMeteredKey(os.Getenv(`UNIDOC_LICENSE_API_KEY`))
+	if err != nil {
+		panic(err)
+	}
+
+	common.SetLogger(common.NewConsoleLogger(common.LogLevelDebug))
+}
+
+func main() {
+	c := creator.New()
+	c.SetPageMargins(30, 30, 20, 20)
+
+	// Read main content template.
+	mainTpl, err := readTemplate("templates/main.tpl")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Read boarding pass data json.
+	pass, err := readBoardingPassData("boarding_pass.json")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Create QR code image.
+	qrCode, err := createQRCode("https://github.com/unidoc/unipdf-examples/tree/master/templates/boarding-pass", 100, 100)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Draw main content teplate.
+	tplOpts := &creator.TemplateOptions{
+		ImageMap: map[string]*model.Image{
+			"qr-code-1": qrCode,
+		},
+		HelperFuncMap: template.FuncMap{
+			"extendDict": func(m map[string]interface{}, params ...interface{}) (map[string]interface{}, error) {
+				lenParams := len(params)
+				if lenParams%2 != 0 {
+					return nil, core.ErrRangeError
+				}
+
+				for i := 0; i < lenParams; i += 2 {
+					key, ok := params[i].(string)
+					if !ok {
+						return nil, core.ErrTypeError
+					}
+
+					m[key] = params[i+1]
+				}
+
+				return m, nil
+			},
+		},
+	}
+
+	if err := c.DrawTemplate(mainTpl, pass, tplOpts); err != nil {
+		log.Fatal(err)
+	}
+
+	// Write output file.
+	if err := c.WriteToFile("unipdf-boarding-pass.pdf"); err != nil {
+		log.Fatal(err)
+	}
+}
+
+// readTemplate reads the template at the specified file path and returns it
+// as an io.Reader.
+func readTemplate(tplFile string) (io.Reader, error) {
+	file, err := os.Open(tplFile)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	buf := bytes.NewBuffer(nil)
+	if _, err = io.Copy(buf, file); err != nil {
+		return nil, err
+	}
+
+	return buf, nil
+}
+
+// createQRCode creates a new QR code image encoding the provided text, having
+// the specified with and height.
```

Comment:
Minor typo in the comment: `with` instead of `width`. I think this is a typo in other examples as well.
Could you fix those as well please?